### PR TITLE
[github] Tests reports

### PR DESCRIPTION
**Pull request by vguerci**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -60,3 +60,16 @@ jobs:
     - name: Test
       run: |
         bundle exec fastlane test
+
+    - name: Test reports upload
+      uses: actions/upload-artifact@master
```
Comment (TurpIF) [inline .github/workflows/main.yml:65]: upload-artifact is now using main ad default branch

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -60,3 +60,16 @@ jobs:
...
+      with:
+        name: test-reports
+        path: "fastlane/test_output"
+
+    - name: Tests reports annotations
+      uses: ashley-taylor/junit-report-annotations-action@1.3
```
Comment (TurpIF) [inline .github/workflows/main.yml:71]: FYI this action is currently broken. This [PR](https://github.com/ashley-taylor/junit-report-annotations-action/pull/16) should fix it.

**Comment by vguerci** [comment]
Removed annotations until It works, fixed the upload version.
Side note, you should avoid targeting @main / @master as it is fragile and could even be, even if unlikely, an attack vector.

**Comment by vguerci** [comment]
Switched to main as it is not v4 specific _(and also because tests are failing on v4 🤔)_

**Pull request merged**